File: dohyeon/Programmers/Lv1_소수_만들기.py
def is_prime_number(x) :
    for i in range(2, x) :
        if x % i == 0 :
            return False
    return True
    
def solution(nums) :
    answer = 0
    for i in range(len(nums)) :
        for j in range(i + 1, len(nums)) :
            for k in range(j + 1, len(nums)) :
                add_number_result = nums[i] + nums[j] + nums[k]
                if is_prime_number(add_number_result) == True :
                    answer += 1
    return answer

print(solution([1,2,3,4]))
print(solution([1,2,7,6,4]))

File: sukjoo/Programmers/Level_1/Lv1_소수_만들기.java
import java.util.*;
class Lv_소수_만들기 {
    public int solution(int[] nums) {
        int answer = 0;
        Arrays.sort(nums);
        for(int i = 0; i < nums.length; i++){
            
            for(int j = i+1; j < nums.length; j++){
                
                for(int m = j+1; m < nums.length; m++){
                    int sum = nums[i] + nums[j] + nums[m];
                    boolean thisSum = true;
                    for(int n = 2; n <= sum/2; n++) {
                        if(sum%n == 0) {
                            thisSum = false;
                            break;
                        }
                    }
                    if(thisSum == true){
                        answer++;
                    }
                }
            }
        }

        return answer;
    }
}